app/javascript/middleware/jdr-component.ts

```diff
@@ -0,0 +1,188 @@
+import * as ng from 'angular';
+import * as _ from 'lodash';
+
+export default class JdrReport implements ng.IComponentOptions {
+  public controller: any = JdrReportController;
+  public templateUrl: string = '/static/middleware/jdr.html.haml';
+  public controllerAs: string = 'mwjdr';
+  public bindings: any = {
+    serverId: '@',
+    downloadUrl: '@',
+    deleteUrl: '@'
+  };
+}
+
+class JdrReportController {
+  public static $inject = ['MiQDataTableService', 'MiQEndpointsService', '$window', '$http'];
+  public serverId: string;
+  public settings: any = {};
+  public rows: any;
+  public cols: any;
+  public perPage: any;
+  public downloadUrl: string;
+  public deleteUrl: string;
+
+  public constructor(private MiQDataTableService, private MiQEndpointsService, private $window, private $http) {}
+
+  public $onInit() {
+    this.initPerPageObject();
+    this.MiQEndpointsService.rootPoint = '/' + ManageIQ.controller;
+    this.MiQEndpointsService.endpoints.listDataTable = '/' + ManageIQ.constants.reportData;
+    this.bindToRx();
+    this.fetchData();
+  }
+
+  private fetchData() {
+    this.settings.isLoading = true;
+    this.rows = [];
+    return this.MiQDataTableService
+      .retrieveRowsAndColumnsFromUrl('ManageIQ::Providers::Hawkular::MiddlewareManager::MiddlewareDiagnosticReport',
+                                     null,
+                                     this.serverId,
+                                     false,
+                                     this.settings)
+      .then(payload => this.onFetchData(payload));
+  }
+
+  private onFetchData(payload) {
+    this.rows = payload.rows;
+    this.cols = [
+      ...payload.cols,
+      {
+        'text': __('Actions'),
+        'sort': 'str',
+        'align': 'left'
+      }
+    ];
+    this.rows.forEach(oneRow => {
+      oneRow.cells = this.addDownloadButton(oneRow);
+    });
+    this.perPage.text = payload.settings.perpage;
+    this.perPage.value = payload.settings.perpage;
+    this.settings = payload.settings;
+    this.settings.dropdownClass = 'dropup';
+    this.settings.selectAllTitle = 'select all';
+    this.settings.columns = this.cols;
+    this.updateSort();
+    this.settings.translateTotalOf = function(start, end, total) {
```

Comment:
Why not use ES6 '=>' instead of **function**

Reply:
Good pint here.